File: src/main/java/no/nav/samordning/hendelser/hendelse/Hendelse.java
package no.nav.samordning.hendelser.hendelse;

import java.time.LocalDate;

public class Hendelse {

    private String ytelsesType;
    private String identifikator;
    private String vedtakId;
    private String fom;
    private String tom;

    public Hendelse(String identifikator, String ytelsesType, String vedtakId, String fom, String tom) {
        this.identifikator = identifikator;
        this.ytelsesType = ytelsesType;
        this.vedtakId = vedtakId;
        this.fom = fom;
        this.tom = null;
    }

    public Hendelse() {}

    public String getYtelsesType() {
    return ytelsesType;
    }

    public void setYtelsesType(String ytelsesType) {
    this.ytelsesType = ytelsesType;
    }

    public String getIdentifikator() {
    return identifikator;
    }

    public void setIdentifikator(String identifikator) {
    this.identifikator = identifikator;
    }

    public String getVedtakId() {
    return vedtakId;
    }

    public void setVedtakId(String vedtakId) {
    this.vedtakId = vedtakId;
    }

    public String getFom() {
    return fom;
    }

    public void setFom(String fom) {
    this.fom = fom;
    }

    public String getTom() {
    return tom;
    }

    public void setTom(String tom) {
    this.tom = tom;
    }
}

File: src/main/kotlin/no/nav/samordning/hendelser/hendelse/Hendelse.kt
package no.nav.samordning.hendelser.hendelse

import com.fasterxml.jackson.annotation.JsonFormat
import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import com.fasterxml.jackson.databind.annotation.JsonSerialize
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateDeserializer
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateSerializer
import no.nav.samordning.hendelser.kafka.SamHendelse
import java.io.Serializable
import java.time.LocalDate

@Suppress("unused")
class Hendelse(
    val ytelsesType: String,
    val identifikator: String,
    val vedtakId: String,
    var samId: String? = null,
    @JsonFormat(pattern="yyyy-MM-dd", timezone = "Europe/Oslo")
    @JsonSerialize(using = LocalDateSerializer::class)
    @JsonDeserialize(using = LocalDateDeserializer::class)
    val fom: LocalDate,
    @JsonFormat(pattern="yyyy-MM-dd", timezone = "Europe/Oslo")
    @JsonSerialize(using = LocalDateSerializer::class)
    @JsonDeserialize(using = LocalDateDeserializer::class)
    var tom: LocalDate? = null
) : Serializable {

    constructor(hendelse: SamHendelse) : this(
        ytelsesType = hendelse.ytelsesType,
        identifikator = hendelse.identifikator,
        vedtakId = hendelse.vedtakId,
        samId = hendelse.samId,
        fom = hendelse.fom,
        tom = hendelse.tom
    )
}

